source/PlayerInfo.cpp

```diff
@@ -2112,6 +2112,28 @@ void PlayerInfo::SelectNext()
 
 
 
+void PlayerInfo::DeselectAllSecondaries()
+{
+	selectedWeapons.clear();
+}
+
+
+
+void PlayerInfo::ToggleAnySecondary(const Outfit *outfit)
+{
+	if(!flagship || flagship->Outfits().empty())
+		return;
+
+	if(selectedWeapons.find(outfit) != selectedWeapons.end())
```

Comment:
You're doing two lookups: Once to find the entry and then again when erasing. If you store the iterator returned by find and use it to erase you avoid one extra lookup.

Reply:
I think this is done?